edb/server/cluster.py

```diff
@@ -163,29 +137,54 @@ def init(self, *, server_settings=None):
 
         self._init(self._pg_cluster)
 
-    def start(self, wait=60, **settings):
-        port = settings.pop('port', None) or self._port
-        if port == 'dynamic':
-            port = find_available_port()
+    def start(self, wait=60, *, port: int=None, **settings):
+        if port is None:
+            port = self._port
 
-        self._effective_port = port
+        if port == 0:
+            cmd_port = 'auto'
+        else:
+            cmd_port = str(port)
 
         extra_args = ['--{}={}'.format(k.replace('_', '-'), v)
                       for k, v in settings.items()]
-        extra_args.append('--port={}'.format(self._effective_port))
+        extra_args.append(f'--port={cmd_port}')
+        if port == 0:
+            extra_args.append('--echo-runtime-info')
```

Comment:
It would be really nice if the server supported passing a fileno here: `--echo-runtime-info=:4:`.  This way we would be able to leave the standard streams undisturbed.